src/core/ack_tracker.c

```diff
@@ -96,6 +96,39 @@ QuicAckTrackerAddPacketNumber(
         !RangeUpdated;
 }
 
+BOOLEAN
+QuicAckTrackerDidHitReorderingThreshold(
```

Comment:
I've been thinking about this logic and I think the following is a better way to do this:
```c++
bool DidHitReorderingThreshold() {

  if (ReorderingThreshold == 0) return false

  LargestUnacked = AckRanges[AckRanges.Count - 1].Max

  for (Index = AckRanges.Count - 1; Index > 0 Index--) {
    SmallestMissing = AckRanges[Index-1].Max + 1
    if (LargestAcked > SmallestMissing) return false
    if (LargestUnacked - SmallestMissing > ReorderingThreshold) return true
  }

  return false
}
```